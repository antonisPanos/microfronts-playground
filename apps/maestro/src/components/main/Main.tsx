import React, {Fragment, FunctionComponent, ReactElement, StrictMode} from 'react';
import {hot} from 'react-hot-loader';
import {
  createHistory,
  History,
  LocationProvider,
  RouteComponentProps,
  Router,
  WindowLocation
} from '@reach/router';
import MicroFront from '../microfront/MicroFront';
import UserDetails from '../user-details/UserDetails';
import {createHashSource} from 'reach-router-hash-history';
import {AppsEnum} from '../../shared/enums';

const Main: FunctionComponent = (): ReactElement => {
  const history: History = createHistory(createHashSource());

  const ReactApp: FunctionComponent<RouteComponentProps<{location: WindowLocation}>> = ({
    location
  }) => <MicroFront app={AppsEnum.REACT_APP} location={location!} />;

  const NgApp: FunctionComponent<RouteComponentProps<{location: WindowLocation}>> = ({
    location
  }) => <MicroFront app={AppsEnum.NG_APP} location={location!} />;

  return (
    <StrictMode>
      <Fragment>
        <div style={{backgroundColor: '#f7f3e9', padding: '10px', borderRadius: '10px'}}>
          <section>
            <h2>Maestro</h2>
            <UserDetails />
            <hr />
            <h2>Navigate</h2>
            <ul>
              <li>
                <a href="#/home">
                  <strong>/home</strong> - (ReactApp)
                </a>
              </li>
              <li>
                <a href="#/user-form">
                  <strong>/user-form</strong> - (ReactApp)
                </a>
              </li>
              <li>
                <a href="#/ng">
                  <strong>/ng</strong> - (NgApp - Home)
                </a>
              </li>
              <li>
                <a href="#/ng-route-info">
                  <strong>/ng-route-info</strong> - (NG App)
                </a>
              </li>
              <li>
                <a href="#/ng-user-form">
                  <strong>/ng-user-form</strong> - (NG App)
                </a>
              </li>
              <li>
                <a href="#/ng-subdomain/ng-route-info">
                  <strong>/ng-subdomain/ng-route-info</strong> - (NG App)
                </a>
              </li>
            </ul>
          </section>
          <hr />
          <section>
            <h2>Router outlet</h2>
            <div
              style={{
                border: '1px dashed black',
                borderRadius: '10px',
                backgroundColor: '#d5ecc2'
              }}
            >
              <LocationProvider history={history}>
                {context => (
                  <Fragment>
                    <Router location={context.location}>
                      <ReactApp location={context.location} path="/home" />
                      <ReactApp location={context.location} path="/user-form" />
                      <NgApp location={context.location} path="/ng" />
                      <NgApp location={context.location} path="/ng-route-info" />
                      <NgApp location={context.location} path="/ng-user-form" />
                      <NgApp location={context.location} path="/ng-subdomain/ng-route-info" />
                    </Router>
                  </Fragment>
                )}
              </LocationProvider>
            </div>
          </section>
        </div>
      </Fragment>
    </StrictMode>
  );
};

declare let module: Record<string, unknown>;

export default hot(module)(Main);
