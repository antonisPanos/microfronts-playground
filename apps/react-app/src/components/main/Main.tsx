import React, {Fragment, FunctionComponent, ReactElement, StrictMode} from 'react';
import {hot} from 'react-hot-loader';
import {Router, WindowLocation} from '@reach/router';
import {UserForm} from '../user-form/UserForm';
import {RouteInfo} from '../route-info/RouteInfo';

const Main: FunctionComponent<{location?: WindowLocation}> = ({
  location = window.location
} = {}): ReactElement => {
  return (
    <StrictMode>
      <Fragment>
        <div style={{padding: '10px', borderRadius: '10px'}}>
          <h1>ReactApp</h1>
          <section>
            <h2>Router outlet</h2>
            <div style={{border: '1px dashed black', padding: '10px', borderRadius: '10px'}}>
              <Router location={location as WindowLocation}>
                <RouteInfo location={location as WindowLocation} path="/home" />
                <UserForm location={location as WindowLocation} path="/user-form" />
              </Router>
            </div>
          </section>
        </div>
      </Fragment>
    </StrictMode>
  );
};

declare let module: Record<string, unknown>;

export default hot(module)(Main);
