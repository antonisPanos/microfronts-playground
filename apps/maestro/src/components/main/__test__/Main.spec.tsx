import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import Main from '../Main';

describe('Main', () => {
  it('should render correctly', () => {
    const mainComp: ShallowWrapper = shallow(<Main />);

    expect(mainComp.debug()).toMatchSnapshot();
  });
});
