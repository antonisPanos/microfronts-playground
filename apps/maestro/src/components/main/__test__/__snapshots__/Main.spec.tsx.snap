// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
"<StrictMode>
  <div style={{...}}>
    <section>
      <h2>
        Maestro
      </h2>
      <UserDetails />
      <hr />
      <h2>
        Navigate
      </h2>
      <ul>
        <li>
          <a href=\\"#/home\\">
            <strong>
              /home
            </strong>
             - (ReactApp)
          </a>
        </li>
        <li>
          <a href=\\"#/user-form\\">
            <strong>
              /user-form
            </strong>
             - (ReactApp)
          </a>
        </li>
        <li>
          <a href=\\"#/ng\\">
            <strong>
              /ng
            </strong>
             - (NgApp - Home)
          </a>
        </li>
        <li>
          <a href=\\"#/ng-route-info\\">
            <strong>
              /ng-route-info
            </strong>
             - (NG App)
          </a>
        </li>
        <li>
          <a href=\\"#/ng-user-form\\">
            <strong>
              /ng-user-form
            </strong>
             - (NG App)
          </a>
        </li>
        <li>
          <a href=\\"#/ng-subdomain/ng-route-info\\">
            <strong>
              /ng-subdomain/ng-route-info
            </strong>
             - (NG App)
          </a>
        </li>
      </ul>
    </section>
    <hr />
    <section>
      <h2>
        Router outlet
      </h2>
      <div style={{...}}>
        <LocationProvider history={{...}}>
          [function]
        </LocationProvider>
      </div>
    </section>
  </div>
</StrictMode>"
`;
